src/TestUtils/CloudFunctionDeploymentTrait.php

```diff
@@ -146,7 +146,7 @@ public function getBaseUri()
      */
     private function processFunctionLogs(string $startTime, callable $process)
     {
-        $projectId = getenv('GOOGLE_CLOUD_PROJECT');
+        $projectId = getenv('GOOGLE_CLOUD_PROJECT') ?: getenv('GOOGLE_PROJECT_ID');
```

Comment:
For this `getenv` call and the one above (in `setUpFunction`) I think it would be better to use `self::$projectId`.  This is defined in `TestTrait::checkProjectEnvVars`, and requires one of `GOOGLE_CLOUD_PROJECT` or `GOOGLE_PROJECT_ID`. That way, this goofy logic is contained in one place.

For the one in `setUpFunction`, it's tricker because the order isn't guaranteed, so it could have this (unfortunate) logic in it:

```php
if (empty(self::$projectId)) {
    self::checkProjectEnvVars();
}
```

Reply:
Done; PTAL?